src/Shouldly.Tests/ConventionTests/ApprovePublicApi.ShouldlyApi.approved.cs

```diff
@@ -818,7 +818,6 @@ public static class ShouldlyConfiguration
         public static double DefaultFloatingPointTolerance;
         public static System.TimeSpan DefaultTaskTimeout;
         public static System.Collections.Generic.List<string> CompareAsObjectTypes { get; }
-        public static Shouldly.Configuration.DiffToolConfiguration DiffTools { get; }
```

Comment:
How will existing code using Shouldly v3 know what to do upon upgrading to v4?

Reply:
Is it possible in principle to preserve Shouldly's existing API and delegate to DiffTool for the implementation?

Reply:
> How will existing code using Shouldly v3 know what to do upon upgrading to v4?

I added obsoletes

> Is it possible in principle to preserve Shouldly's existing API and delegate to DiffEngine for the implementation

i would prefer not to. reasons:

 * many of the diff APIs should not have been made public. eg KnownDoNotLaunchStrategies, DoNotLaunchWhenPlatformIsNotWindows
 * there is a mismatch between the APIs. for example shouldly is hard coded to expect that a single tool has the same args for different OS. this is not correct eg https://github.com/VerifyTests/DiffEngine/blob/master/docs/diff-tool.md#windows-settings so that cant be properly represented in the existing shouldly api. The may be other examples i have not found, and future changes in DiffEngine could result in more of a mismatch
 * most of the shouldly Diff api should be very rarely used required by people. for example given the [tools supported by DiffEngine](https://github.com/VerifyTests/DiffEngine#supported-tools) people should rarely need to add a custom tool
 * shouldly lack opinion on if settings should be (personal or shared)[https://github.com/VerifyTests/DiffEngine/blob/master/docs/code-verus-machine-settings.md]
 * the shouldly API docs would need be continually maintained

Reply:
I have toyed with the idea of avoiding a nuget ref. submodules or a source package. but ultimately the docs was too difficult to replicate

Reply:
thoughts on the above? this PR changes enough that i am holding of on the next few things

Reply:
That answers my question, thanks. I was thinking it could ease the transition, but if we want the obsoleted APIs to continue working in v4 we should just leave the v3 impl intact.

I haven't used diffing so I don't have a good sense of what people might expect.